etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/TransformationRule.java

```diff
@@ -0,0 +1,86 @@
+package gov.hhs.cdc.trustedintermediary.etor.ruleengine.transformation;
+
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.FhirResource;
+import gov.hhs.cdc.trustedintermediary.etor.ruleengine.Rule;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * The TransformationRule class extends the {@link
+ * gov.hhs.cdc.trustedintermediary.etor.ruleengine.Rule Rule} class and represents a transformation
+ * rule. It implements the {@link
+ * gov.hhs.cdc.trustedintermediary.etor.ruleengine.Rule#runRule(FhirResource) runRule} method to
+ * apply a transformation to the FHIR resource.
+ */
+public class TransformationRule extends Rule<TransformationRuleMethod> {
+
+    private static final Map<String, Class<?>> classCache = new ConcurrentHashMap<>();
+
+    /**
+     * Do not delete this constructor! It is used for JSON deserialization when loading rules from a
+     * file.
+     */
+    public TransformationRule() {}
+
+    public TransformationRule(
+            String ruleName,
+            String ruleDescription,
+            String ruleMessage,
+            List<String> ruleConditions,
+            List<TransformationRuleMethod> ruleActions) {
+        super(ruleName, ruleDescription, ruleMessage, ruleConditions, ruleActions);
+    }
+
+    public void runRule(FhirResource<?> resource) {
+        this.getRules().forEach((transformation -> applyTransformation(transformation, resource)));
+    }
+
+    static Class<?> loadClassFromCache(String className) throws ClassNotFoundException {
+        return classCache.computeIfAbsent(className, TransformationRule::loadClassByName);
+    }
+
+    static void executeCustomTransformationMethod(
+            Class<?> clazz, FhirResource<?> resource, Map<String, String> args)
+            throws NoSuchMethodException, InvocationTargetException, InstantiationException,
+                    IllegalAccessException {
+        Method method = clazz.getDeclaredMethod("transform", FhirResource.class, Map.class);
```

Comment:
Thoughts on using an interface and concrete implementations for each type of transformation instead of reflection?

Reply:
Good point. We already have an interface so I changed it to use that